rust/toolchain.bzl

```diff
@@ -198,10 +202,18 @@ def _make_libstd_and_allocator_ccinfo(ctx, rust_std, allocator_library):
             transitive = [between_core_and_std_inputs],
             order = "topological",
         )
+        test_inputs = depset(
+            [
+                _ltl(f, ctx, cc_toolchain, feature_configuration)
+                for f in rust_stdlib_info.test_files
+            ],
+            transitive = [std_inputs],
+            order = "topological",
+        )
 
         link_inputs = cc_common.create_linker_input(
             owner = rust_std.label,
-            libraries = std_inputs,
+            libraries = test_inputs,
```

Comment:
Feels minor but should we instead track `link_inputs` and `test_link_inputs` separately?

Reply:
I considered this a bit, but I think the way it is now is more consistent with the treatment of other groups of inputs above (e.g., `between_code_and_std_inputs` is immediately folded as a transitive dep of `std_inputs` and not mentioned afterwards). Also,  if we separate them here, we'd have to merge them at some later point, and then the merged result will effectively be the same as `test_inputs` since `std_inputs` is already a transitive dep of `test_inputs`.

Reply:
Sounds reasonable enough, we can split them later if it becomes an issue I suppose. Thanks again!